src/Foundation/NSUrlSessionHandler.cs

```diff
@@ -536,84 +536,97 @@ protected override async Task<HttpResponseMessage> SendAsync (HttpRequestMessage
 
 #if NET
 		// Properties that will be called by the default HttpClientHandler
+
+		// NSUrlSession handler automatically handles decompression, and there doesn't seem to be a way to turn it off.
+		// The available decompression algorithms depend on the OS version we're running on, and maybe the target OS version as well,
+		// so just say we're doing them all, and not do anything in the setter (it doesn't seem to be configurable in NSUrlSession anyways).
 		public DecompressionMethods AutomaticDecompression {
-			get => throw new PlatformNotSupportedException ();
-			set => throw new PlatformNotSupportedException ();
+			get => DecompressionMethods.All;
+			set { }
 		}
 
-		public bool CheckCertificateRevocationList {
-			get => throw new PlatformNotSupportedException ();
-			set => throw new PlatformNotSupportedException ();
-		}
+		// We're ignoring this property, just like Xamarin.Android does:
+		// https://github.com/xamarin/xamarin-android/blob/09e8cb5c07ea6c39383185a3f90e53186749b802/src/Mono.Android/Xamarin.Android.Net/AndroidMessageHandler.cs#L158
+		public bool CheckCertificateRevocationList { get; set; } = false;
 
-		public X509CertificateCollection ClientCertificates {
-			get => throw new PlatformNotSupportedException ();
-		}
+		// We're ignoring this property, just like Xamarin.Android does:
+		// https://github.com/xamarin/xamarin-android/blob/09e8cb5c07ea6c39383185a3f90e53186749b802/src/Mono.Android/Xamarin.Android.Net/AndroidMessageHandler.cs#L150
+		public X509CertificateCollection ClientCertificates { get { return null; } }
```

Comment:
I think this violates the nullability contract. We don't think we'd ever expect getting a null here.
https://github.com/dotnet/runtime/blob/103fb847f66f53557b0e3179599365b65de02ce4/src/libraries/System.Net.Http/src/System/Net/Http/HttpClientHandler.cs#L255-L256
It does look like it could throw if options aren't manual:
https://github.com/dotnet/runtime/blob/103fb847f66f53557b0e3179599365b65de02ce4/src/libraries/System.Net.Http/src/System/Net/Http/HttpClientHandler.cs#L252